packages/compass-explain-plan/src/components/explain-tree/tree-layout.tsx

```diff
@@ -64,11 +74,29 @@ function LinkPath<T>({
     const sourceBottomY = sourceY + actualSourceYSize;
     const elbowY = sourceBottomY + (targetY - sourceBottomY) / 2;
 
-    return `M ${linkStartX} ${linkStartY} V ${elbowY} H ${linkEndX} V ${linkEndY}`;
+    return [
+      `M ${linkStartX} ${linkStartY} V ${elbowY} H ${linkEndX} V ${linkEndY}`,
+    ];
   }, [gapY, link.source, link.target, translateX]);
 
   return (
-    <path fill="none" stroke={linkColor} strokeWidth={linkWidth} d={pathDef} />
+    <>
+      <path
+        fill="none"
+        stroke={linkColor}
+        strokeWidth={linkWidth}
+        d={pathDef}
+      />
+      {isLastLink && (
```

Comment:
I haven't looked close on how it's implemented, but doesn't look like this is working as expected. Only one card gets the arrow, where based on the designs it seems like we expect all leaf nodes to get it

![image](https://github.com/mongodb-js/compass/assets/5036933/a2bfd3d3-168c-4482-af69-bf4e3d47f076)